src/minichain/network.py

```
@@ -0,0 +1,1075 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import json
+import secrets
+import socket
+import struct
+import time
+from collections import deque
+from dataclasses import asdict, dataclass, field
+from typing import Any, Callable, Coroutine
```

Ruff (0.15.2)

[warning] Import from `collections.abc` instead: `Callable`, `Coroutine`

Import from `collections.abc`

(UP035)